distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -809,9 +809,21 @@ type Column
              example_truncate = Examples.decimal_column.truncate
     truncate : Column ! Invalid_Value_Type
     truncate self =
-        Value_Type.expect_numeric self <| self.short_circuit_special_floating_point <|
-            new_name = self.naming_helper.function_name "truncate" [self]
-            self.make_unary_op "TRUNCATE" new_name
+        new_name = self.naming_helper.function_name "truncate" [self]
+        precise_value_type = self.inferred_precise_value_type
+        case precise_value_type.is_numeric of
+            True ->
+                self.short_circuit_special_floating_point <|
+                    self.make_unary_op "TRUNCATE" new_name
+            False -> case precise_value_type == Value_Type.Date_Time of
+                True ->
+                    op = "date_trunc_to_day"
+                    case self.connection.dialect.is_supported op of
+                        True ->
+                            self.make_unary_op op new_name
+                        False ->
+                            Error.throw (Unsupported_Database_Operation.Error ("`Column.truncate` for `Date_Time` is not supported by this connection."))
```

Comment:
Technically, this is dead code currently, because Postgres has this operation and in SQLite we cannot reach here, because first we do a check `precise_value_type == Value_Type.Date_Time` - which will never be true in SQLite (currently) as it does not have date-time columns.

But I think it's good to have it - it will make for clearer errors once we add date-time to SQLite or add another backend and forget to implement truncate for it.

👍